src/libs/SearchQueryUtils.ts

```diff
@@ -96,19 +95,21 @@ function sanitizeSearchValue(str: string) {
 function buildDateFilterQuery(filterValues: Partial<SearchAdvancedFiltersForm>, filterKey: SearchDateFilterKeys) {
     const dateBefore = filterValues[`${filterKey}${CONST.SEARCH.DATE_MODIFIERS.BEFORE}`];
     const dateAfter = filterValues[`${filterKey}${CONST.SEARCH.DATE_MODIFIERS.AFTER}`];
+    const dateOn = filterValues[`${filterKey}${CONST.SEARCH.DATE_MODIFIERS.ON}`];
+
+    const dateFilters = [];
 
-    let dateFilter = '';
     if (dateBefore) {
-        dateFilter += `${filterKey}<${dateBefore}`;
-    }
-    if (dateBefore && dateAfter) {
-        dateFilter += ' ';
+        dateFilters.push(`${filterKey}<${dateBefore}`);
     }
     if (dateAfter) {
-        dateFilter += `${filterKey}>${dateAfter}`;
+        dateFilters.push(`${filterKey}>${dateAfter}`);
+    }
+    if (dateOn) {
+        dateFilters.push(`${filterKey}:${dateOn}`);
     }
 
-    return dateFilter;
+    return dateFilters.join(' ');
```

Comment:
Let's leave this for a follow up since this PR is already pretty large, but we need to join the `dateOn` filter with an `OR` so that a query like `date<2025-05-06 date>2025-05-01 OR date:2025-05-10` works properly. I just realized that we don't seem to be parsing `OR` operators in Search at all. So that's why I say we handle it in a follow up